File: Main/DailyLeetcode/Group/566.cc
#include <vector>
using namespace std;

class Solution {
  public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &mat, int r, int c) {
        vector<vector<int>> newShape(r, vector<int>(c, 0));
        int orows = mat.size();
        int oc = mat[0].size();

        if (orows * oc != r * c) return mat;
        r--;
        c--;
        int c2 = c;
        int r2 = r;
        for (int i = orows - 1; i >= 0; i--) {
            for (int j = oc - 1; j >= 0; j--) {
                if (c2 < 0) {
                    c2 = c;
                    r2--;
                }

                if (r2 < 0) {
                    break;
                }
                newShape[r][c] = mat[i][j];
                c--;
            }
        }
        return newShape;
    }
};

File: gca3/566.py
from typing import List

class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        n, m = len(mat), len(mat[0])
        if n * m != r * c:
            return mat
        matrix = [[0 for _ in range(c)] for _ in range(r)]
        x, y = 0, 0
        

        for i in range(n):
            for j in range(m):
                if y >= c:
                    y = 0
                    x+= 1
                matrix[x][y] = mat[i][j]
                y += 1

        return matrix